### PR TITLE
Implement stop_listening

### PR DESCRIPTION
**Pull request by asvetlov**
Close all listening connection to stop accepting new clients.
Part of graceful shutdown procedure.

### CODE REVIEW

File: aiohttp/helpers.py [hunk 1]
```diff
@@ -514,3 +515,22 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 
     def _cancel_task(self):
         self._cancelled = self._task.cancel()
+
+
+def stop_listening(srv, loop):
```
Comment (redixin) [inline aiohttp/helpers.py:520]: This looks weird. I thought is enough to call `.close()` method of `protocol` instance.

File: aiohttp/helpers.py [hunk 2]
```diff
@@ -514,3 +515,22 @@ def __exit__(self, exc_type, exc_val, exc_tb):
...
+    """Stop listening Server instance.
+
+    Close all listening sockets leaving accepted connections untouched.
+    """
+    to_remove = []
+    for sock in srv.sockets:
+        try:
+            if sock.getsockopt(socket.SOL_SOCKET, socket.SO_ACCEPTCONN):
+                loop.remove_reader(sock.fileno())
```
Comment (fafhrd91) [inline aiohttp/helpers.py:529]: is `remove_reader` available for every loop type?

File: aiohttp/helpers.py (hunk 1 above)
Comment (redixin) [inline aiohttp/helpers.py:520]: Sorry, I mean `srv` instance, return value of `loop.create_server`.

``` python
srv = loop.create_server(AiohttpHttpProtocol, ....
#
srv.close()  # stop listening
await wait_for_connected_clients(timeout=timeout) # graceful
await srv.wait_closed() # close remainig
``
```

File: aiohttp/helpers.py (hunk 2 above)
Comment (asvetlov) [inline aiohttp/helpers.py:529]: Good point. It's not supported by Windows IOCP.
BaseEventLoop has `_stop_serving(sock)` method but I don't want to use private API.

Comment (redixin) [inline aiohttp/helpers.py:529]: @asvetlov 
Sorry, but I still can't get what is going on here. `aiohttp.ServerHttpProtocol` is subclass of `asyncio.Protocol`, and `srv.sockets` contain only listening sockets, and `srv.close()` does the "stop listening" thing.

Comment (asvetlov) [inline aiohttp/helpers.py:529]: Perhaps you are right. I need double check.

**Comment by asvetlov** [comment]
@redixin thanks

**Pull request closed without merging**